marlowe-runtime/test/Language/Marlowe/Runtime/History/Script.hs

```diff
@@ -193,29 +198,34 @@ reduceScriptEvent = \case
                     { contractVersion
                     , stateScriptOutput
                     , statePayouts = Map.delete utxo statePayouts
+                      , stateScriptAddress
+                      , statePayoutAddress
                     }
                   state' = state { contractStates = Map.insert contractId (Some contractState') contractStates }
                 in
                   Right $ state' : states
 
-applyInputsV1 :: P.POSIXTime -> P.POSIXTime -> TxId -> [V1.Input] -> V1.MarloweData -> Maybe (TransactionOutput 'V1)
-applyInputsV1 vLow vHigh txId txInputs V1.MarloweData{..} =
+applyInputsV1 :: Address -> Address -> P.POSIXTime -> P.POSIXTime -> TxId -> [V1.Input] -> V1.MarloweData -> Maybe (TransactionOutput 'V1)
```

Comment:
Either here or somewhere else before `computeTransaction` is called, we need to check and possibly adjust the POSIX time range to account for the node's ouroboros conversion of slot to time and the Marlowe validator's use of `fixInterval`. Otherwise we will see perplexing validation failures. Meticulous attention to time/slot conversion consistency is critically important.

See [SCP-4461](https://input-output.atlassian.net/browse/SCP-4461) for requirements plus solution suggestion, and PR #269 for work in progress.

CC: @paluh

Reply:
We have an implementation based on slot rounding but @bwbush suspects that it can be buggy or problematic in the `marlowe-cli`.
The conversion itself is a part of the implementation of `buildApplyInputsConstraints` (I added a reference to this discussion in the ticket acceptance criteria) so we can discuss the details there.